File: LibLink_API/LibLink_API/Models/Reservation.cs
namespace LibLink_API.Models
{
    public class Reservation
    {
        public int ReservationID { get; set; }

        public int UserID { get; set; }

        public int BookID { get; set; }

        public DateTime ReservationDate { get; set; }

        public string Status { get; set; } = "Waiting";

        public User User { get; set; } = null!;

        public Book Book { get; set; } = null!;
    }
}

File: app/src/main/java/com/example/liblinkapp/models/Reservation.kt
package com.example.liblinkapp.models

import com.google.gson.annotations.SerializedName

data class Reservation(

    @SerializedName("reservationID")
    val id: Int,

    @SerializedName("userID")
    val userId: Int,

    @SerializedName("bookID")
    val bookId: Int,

    val reservationDate: String?,

    val status: String,

    val book: Book?
)